cluster-autoscaler/core/scale_up.go

```diff
@@ -285,6 +285,10 @@ func ScaleUp(context *context.AutoscalingContext, processors *ca_processors.Auto
 
 	upcomingNodes := make([]*schedulercache.NodeInfo, 0)
 	for nodeGroup, numberOfNodes := range clusterStateRegistry.GetUpcomingNodes() {
+		if !clusterStateRegistry.IsNodeGroupHealthy(nodeGroup) {
+			klog.Warningf("Node group %s has %d upcoming nodes but is unhealthy. Ignoring those nodes.", nodeGroup, numberOfNodes)
```

Comment:
Even if 80% of the group is broken, but some nodes are just being started, we should not ignore them. They have a chance of being healthy and should be included in the upcoming nodes for further evaluation.

Reply:
`upcomingNodes` is only being used to evaluate if a nodeGroup should be expanded. The assumption being that all the upcoming nodes are indeed upcoming, healthy nodes.

The problem is that in some cases, like with spot instances in AWS, the nodes will never show up. So the worst case scenario is that a pod will never be scheduled.
If upcoming nodes from a broken nodegroup do end up showing up healthy, the worst case scenario is that we have over-provisioned the cluster and it will need to scale down in the next possible ScaleDown event.

Reply:
The fact that nodegroup is unhealthy doesn't imply that the upcoming nodes won't turn up ok. They may or may not and your solution is trading one problem for the other. 

There is already a mechanism in CA for handling your case - if we know the nodes will never show up, NodeGroup.Nodes() should return Instances specifying InstanceState=InstanceCreating and appropriate ErrorInfo. If  CA gets such Instances from cloudprovider it will delete those VMs immediately, backoff nodegroup and trigger a scale-up in a different nodegroup. I think the right way to fix your issue would be to improve implementation of NodeGroup.Nodes() in AWS cloudprovider, so that it's actually returns Instances with appropriate error.

Reply:
@viggeh Please fix the problem in the way @MaciekPytel described and submit a new PR. I'm closing this PR as the right solution will be significantly different.

Reply:
Thank you. I'll take a look at that approach.